src/execute/oas3/style-serializer.js

```diff
@@ -59,10 +43,14 @@ export default function stylize(config) {
 export function valueEncoder(value, escape) {
```

Comment:
If we look at the current code of value encoder:

```js
export function valueEncoder(value, escape) {
  if (Array.isArray(value) || (value !== null && typeof value === 'object')) {
    value = JSON.stringify(value);
  } else if (typeof value === 'number' || typeof value === 'boolean') {
    value = value.toString();
  }

  if (escape && value.length > 0) {
    return encodeCharacters(value, escape === true ? {} : { characterSet: escape });
  }
  return value;
}
```

I can conclude multiple observations:

### 1. the valueEncoder doesn't encode characters unless explicitly requested

I would suggest to change the signature to 

```js
export function valueEncoder(value, escape = false) {
```

### 2. value encoder assumes that the `escape` can be `true`

We've established that the allowed values for the `escape` parameter can be: `false`, `reserved` and `unreserved`

```js
return encodeCharacters(value, escape === true ? {} : { characterSet: escape });
```

So the above code passes `true` to the `encodeCharacters`, which is wrong. It should just omit the `characterSet` option so that default characterSet will kick in. 

Still valueEncoder itself should not receive `true` and `escape` value.